src/manipulation-planner.cc

```diff
@@ -445,10 +445,6 @@ namespace hpp {
             continue;
           bool _1to2 = (*itn1)->isOutNeighbor (*itn2);
           bool _2to1 = (*itn1)->isInNeighbor (*itn2);
-          if (_1to2 && _2to1) {
-            hppDout (info, "the two nodes are already connected");
-            continue;
-          }
```

Comment:
Is it always false by design of the algorithm or in your experiments ?

I would keep it as an assert. I don't remember why I introduced it. For reference, it was added in 2065178 and it is related to 7d1cb92

Reply:
I would say it is false by design:
 - itn2 is in itcc
 - if _1to2 and _2to1, then itn2 and itn1 are in the same connected component: itcc. This is not possible since the test above should have continued:
```
          if (*itcc == (*itn1)->connectedComponent ())
            continue;
```
But we can add an assert. Although asserts are called only in debug mode.

Reply:
It is false by design. But I will put an assert.
I have tested with the benchmark. I will also handle the test in `tryConnectToRoadmap`.

Reply:
I agree the test should be false by design. I let you decide on whether to keep the assert. In any case, the test is here only to save some computations.